clustered/integration-test/src/test/java/org/ehcache/clustered/util/ParallelTestCluster.java

```diff
@@ -27,7 +28,7 @@
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 
-public class ParallelTestCluster extends Cluster {
+public class ParallelTestCluster implements TestRule {
```

Comment:
This has decoupled `ParallelTestCluster` from `Cluster`. This seems like bad OO and is unhelpful from a maintenance POV. Ideally I would like to have seen `Cluster` decoupled from JUnit entirely in the earlier platform PR and then have this class implement both `Cluster` and `TestRule` independently.

Reply:
Yes this is right we should go even further 👍 
This is another change to do then. 
Could you please open an issue in tc-platform describing that and an assign @saurabhagas to it ? 
We can track that in our board in our user story regarding QA/Testing, with low priority.
This is something he could work when I will be out.